addons/knowledge/controllers/main.py

```diff
@@ -0,0 +1,319 @@
+            'parent_permission': article.parent_id.inherited_permission,
+            'based_on': article.inherited_permission_parent_id.display_name,
+            'based_on_id': article.inherited_permission_parent_id.id,
+            'members_options': permission_field.get_description(request.env).get('selection', []),
+            'members': members_values,
+            'is_sync': is_sync,
+            'parent_id': article.parent_id.id,
+            'parent_name': article.parent_id.display_name,
+            'user_is_admin': user_is_admin,
+            'show_admin_tip': user_is_admin and article.user_permission != 'write',
+        }
+
+    @http.route('/knowledge/article/set_member_permission', type='json', auth='user')
+    def article_set_member_permission(self, article_id, permission, member_id=False, partner_id=False, **post):
+        """
+        Sets the permission of the given member for the given article.
+        **Note**: The user needs "write" permission to change the permission of a user.
+        :param article_id: (int) article id
+        :param permission: (string) permission
+        :param member_id: (int) member id
+        :param partner_id: (int) partner id. Typically given when the member is based on a parent article.
+        """
+        article = request.env['knowledge.article'].sudo().browse(article_id)
+        if not article.user_can_write:
+            return werkzeug.exceptions.Forbidden()
+        previous_category = article.category
+        if partner_id:  # If based on
+            partner = request.env['res.partner'].browse(partner_id).exists()
+            if not article or not partner:
+                return {'success': False}
+            member = request.env['knowledge.article.member'].browse(member_id).exists()
+            downgrade = bool(member) and member.permission > permission
+            if (not downgrade and not article.invite_members(partner, permission, send_mail=False)) \
+                    or (downgrade and not article._desync_access_from_parents(partner.ids, permission)):
+                return {'success': False}
+        elif not article or not article._set_member_permission(member_id, permission):
+            return {'success': False}
+        result = {'success': True}
+        if article.category != previous_category:
+            result['reload_tree'] = True
+        return result
+
+    @http.route('/knowledge/article/remove_member', type='json', auth='user')
+    def article_remove_member(self, article_id, member_id, partner_id=False):
+        """
+        Removes the given member from the given article.
+        The function returns a dictionary indicating whether the request succeeds (see: `success` key).
+        The dictionary can also include a `reload_tree` entry that will signify the caller that the aside block
+        listing all articles should be reloaded. This can happen when the article moves from one section to another.
```

Comment:
Funny how this methods specified this and not the "set_member_permission" one ^^